### PR TITLE
Using unittest runner

### PR DESCRIPTION
**Pull request by benwatson528**
- Removed dependency on Pytest
- Tests run predictably from any directory

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 name: Python application
 
-on: [push]
+on: [push, pull_request]
```
Comment (benwatson528) [inline .github/workflows/build.yml:6]: Now the Action is attached to the PR as well, so you can see if a PR builds correctly from the PR itself.

image

File: .gitignore [hunk 2]
```diff
@@ -141,4 +141,4 @@ dmypy.json
 # Cython debug symbols
 cython_debug/
 
-/test/csv/line of sight analysis.kml
+/test/data/line_of_sight_analysis.kml
```
Comment (benwatson528) [inline .gitignore:144]: I renamed this folder from `csv` to `data` to avoid conflicts with the `csv` library.

File: requirements.txt [hunk 3]
```diff
@@ -6,4 +6,3 @@ OSGridConverter~=0.1.3
 pyproj~=2.6.1.post1
 haversine~=2.2.0
 mock~=4.0.2
-pytest~=5.4.3
```
Comment (benwatson528) [inline requirements.txt:9]: Bye Pytest.

File: test/test_kml_generator.py [hunk 4]
```diff
@@ -26,3 +29,7 @@ def test_create_linestrings(self):
         location_list_2 = [(Location(54.54, -1.45, 150, "point1_2")), (Location(54.54, -1.44, 150, "point2_2"))]
         linestring_list = create_linestrings(location_list, location_list_2, line_of_sight_folder)
         self.assertEqual("point1, point1_2", str(linestring_list[0].name))
+
+    @staticmethod
+    def create_test_file_path():
+        return os.path.join(CURRENT_DIR, os.pardir, "test/data/")
```
Comment (benwatson528) [inline test/test_kml_generator.py:35]: Previously we were referring to test files by just `csv/filename.csv`, which only works if you're running from the `test` folder. We want to be able to run our test classes from any location and have them figure out where the data files are. This method finds the path to the running file (so `D:/Ben/Git/line-of-sight-analysis/test/test_kml_generator.py`) and then navigates to the data directory relative to that (by sticking `data/` to the end of it).

File: main/kml_generator.py [hunk 5]
```diff
@@ -15,7 +15,7 @@ def create_kml_file(first_location_list, second_location_list, first_folder_name
     first_location_points = create_points(first_location_list, kml_folder_list_one)
     second_location_points = create_points(second_location_list, kml_folder_list_two)
     linestrings = create_linestrings(first_location_list, second_location_list, kml_folder_linestring)
-    save_string = output_folder + "/line of sight analysis.kml"
+    save_string = output_folder + "/line_of_sight_analysis.kml"
```
Comment (benwatson528) [inline main/kml_generator.py:18]: It would still be good to have this go into a dedicated output dir separate from the input data files.

**Comment by benwatson528** [comment]
Good tasks to do next:

- Fix the Lint warnings
- Move the `line_of_sight_analysis.kml` file into its own dir and have the entire dir be ignored.
- Consider adding a formatting action such as [Black](https://github.com/marketplace/actions/black-code-formatter) -  see [here](https://black.readthedocs.io/en/stable/editor_integration.html#pycharm-intellij-idea) for integrating it with PyCharm (it can even be set up to automatically apply every time a file is changed).

File: test/test_kml_generator.py [hunk 6]
```diff
@@ -26,3 +29,7 @@ def test_create_linestrings(self):
         location_list_2 = [(Location(54.54, -1.45, 150, "point1_2")), (Location(54.54, -1.44, 150, "point2_2"))]
         linestring_list = create_linestrings(location_list, location_list_2, line_of_sight_folder)
         self.assertEqual("point1, point1_2", str(linestring_list[0].name))
+
+    @staticmethod
+    def create_test_file_path():
+        return os.path.join(CURRENT_DIR, "data/")
```
Comment (benwatson528) [inline test/test_kml_generator.py:35]: Previously we were referring to test files by just csv/filename.csv, which only works if you're running from the test folder. We want to be able to run our test classes from any location and have them figure out where the data files are. This method finds the path to the running file (so D:/Ben/Git/line-of-sight-analysis/test/test_kml_generator.py) and then navigates to the data directory relative to that (by sticking data/ to the end of it).

**Review by MHenderson1988 — APPROVED** [review]

**Comment by MHenderson1988** [comment]
Thanks Ben

**Pull request merged**